spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -1,1900 +1,2549 @@
-              'failed test precondition:
-              pending enrollments must not set status check attempted time',
-            )
-
-            expect(pending_enrollments.pluck(:status_check_completed_at)).to(
-              all(eq nil),
-              'failed test precondition:
-              pending enrollments must not set status check completed time',
-            )
-
-            freeze_time do
-              job.perform(Time.zone.now)
-
-              expect(
-                pending_enrollments.
-                  map(&:reload).
-                  pluck(:status_check_attempted_at),
-              ).to(
-                all(eq Time.zone.now),
-                'job must update status check attempted time for all pending enrollments',
-              )
+              it 'updates the enrollment status check timestamps' do
+                expect(enrollment.reload).to have_attributes(
+                  status_check_attempted_at: current_time,
+                  status_check_completed_at: current_time,
+                  last_batch_claimed_at: current_time,
+                )
+              end
 
-              expect(
-                pending_enrollments.
-                  map(&:reload).
-                  pluck(:status_check_completed_at),
-              ).to(
-                all(eq Time.zone.now),
-                'job must update status check completed time for all pending enrollments',
-              )
+              it 'logs the job completed analytic' do
+                expect(analytics).to have_received(
+                  :idv_in_person_usps_proofing_results_job_completed,
+                ).with(
+                  **default_job_completion_analytics,
+                  enrollments_checked: 1,
+                  enrollments_in_progress: 1,
+                )
+              end
             end
-          end
 
-          it 'logs a message when the job starts' do
-            enrollment_records = InPersonEnrollment.where(id: pending_enrollments.map(&:id))
-            allow(InPersonEnrollment).to receive(:needs_usps_status_check).
-              and_return(enrollment_records)
-            stub_request_proofing_results_with_responses(
-              request_failed_proofing_results_args,
-              request_in_progress_proofing_results_args,
-              request_in_progress_proofing_results_args,
-              request_failed_proofing_results_args,
-            )
-
-            job.perform(Time.zone.now)
-
-            expect(job_analytics).to have_logged_event(
-              'GetUspsProofingResultsJob: Job started',
-              enrollments_count: 5,
-              reprocess_delay_minutes: 2.0,
-              job_name: 'GetUspsProofingResultsJob',
-            )
-          end
+            context 'when the exception response message is IPP_EXPIRED_ERROR' do
+              before do
+                stub_request_expired_id_ipp_proofing_results
+              end
 
-          it <<~STR.squish do
```

Comment:
@shanechesnutt-ft  This test checked that analytics correctly summarizes various outcomes. I know each of your specs includes `logs the job completed analytic` but do you think there is added value in this spec?

Reply:
I think it could be nice to add this to the spec where I run perform with multiple enrollments. That way we can check to ensure calculations are correct with multiple enrollments during the job.

Reply:
Nice addition! I can see you added this in context `when multiple pending InPersonEnrollments exist` and specifically inside `logs the job completed analytic`. I think having coverage around totals and percentages is a good idea. Resolving comment now.